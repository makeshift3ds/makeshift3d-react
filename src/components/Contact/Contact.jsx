import React from 'react';
import { Markup } from 'interweave';
import 'babel-polyfill';
import './Contact.css';

export default class Home extends React.Component {
  async componentDidMount() {
    try {
      const res = await fetch('http://localhost:3000/pages/contact');
      const page = await res.json();
      this.setState({
        page,
      });
    } catch (e) {
      console.log('error', e);
    }
  }

  state = {
    page: {},
  };

  render() {
    const { title, body } = this.state.page;

    return (
      <div className="content-container">
        <article>
          <h1>{title}</h1>
          <Markup content={body} tag="div" />
        </article>
        <aside>
          <div className="aside-block">
            <p>I know all the twitters.</p>
            <span> - Ken Elliott</span>
          </div>
        </aside>
      </div>
    );
  }
}
