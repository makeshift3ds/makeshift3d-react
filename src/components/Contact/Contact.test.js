import React from 'react';
import { shallow } from 'enzyme';
import Contact from './Contact.jsx';

describe('<Contact />', () => {
  it('should render 1 <Contact /> component', () => {
    const component = shallow(<Contact />);
    expect(component).toHaveLength(1);
  });

  it('should render an article element', () => {
    const component = shallow(<Contact />);
    expect(component.find('article')).toHaveLength(1);
  });

  it('should render an aside element', () => {
    const component = shallow(<Contact />);
    expect(component.find('aside')).toHaveLength(1);
  });

  it('should match snapshot', () => {
    const component = shallow(<Contact />);
    expect(component).toMatchSnapshot();
  });
});
