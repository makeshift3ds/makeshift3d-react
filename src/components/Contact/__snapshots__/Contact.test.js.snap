// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <article>
          <h1 />
          <Markup
            content=""
            disableLineBreaks={false}
            disableWhitelist={false}
            emptyContent={null}
            noHtml={false}
            noHtmlExceptMatchers={false}
            tag="div"
            tagName="span"
          />
        </article>,
        <aside>
          <div
            className="aside-block"
          >
            <p>
              I know all the twitters.
            </p>
            <span>
               - Ken Elliott
            </span>
          </div>
        </aside>,
      ],
      "className": "content-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1 />,
            <Markup
              content=""
              disableLineBreaks={false}
              disableWhitelist={false}
              emptyContent={null}
              noHtml={false}
              noHtmlExceptMatchers={false}
              tag="div"
              tagName="span"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "content": "",
              "disableLineBreaks": false,
              "disableWhitelist": false,
              "emptyContent": null,
              "noHtml": false,
              "noHtmlExceptMatchers": false,
              "tag": "div",
              "tagName": "span",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "article",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="aside-block"
          >
            <p>
              I know all the twitters.
            </p>
            <span>
               - Ken Elliott
            </span>
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                I know all the twitters.
              </p>,
              <span>
                 - Ken Elliott
              </span>,
            ],
            "className": "aside-block",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "I know all the twitters.",
              },
              "ref": null,
              "rendered": "I know all the twitters.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " - Ken Elliott",
              },
              "ref": null,
              "rendered": " - Ken Elliott",
              "type": "span",
            },
          ],
          "type": "div",
        },
        "type": "aside",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article>
            <h1 />
            <Markup
              content=""
              disableLineBreaks={false}
              disableWhitelist={false}
              emptyContent={null}
              noHtml={false}
              noHtmlExceptMatchers={false}
              tag="div"
              tagName="span"
            />
          </article>,
          <aside>
            <div
              className="aside-block"
            >
              <p>
                I know all the twitters.
              </p>
              <span>
                 - Ken Elliott
              </span>
            </div>
          </aside>,
        ],
        "className": "content-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1 />,
              <Markup
                content=""
                disableLineBreaks={false}
                disableWhitelist={false}
                emptyContent={null}
                noHtml={false}
                noHtmlExceptMatchers={false}
                tag="div"
                tagName="span"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "content": "",
                "disableLineBreaks": false,
                "disableWhitelist": false,
                "emptyContent": null,
                "noHtml": false,
                "noHtmlExceptMatchers": false,
                "tag": "div",
                "tagName": "span",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="aside-block"
            >
              <p>
                I know all the twitters.
              </p>
              <span>
                 - Ken Elliott
              </span>
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  I know all the twitters.
                </p>,
                <span>
                   - Ken Elliott
                </span>,
              ],
              "className": "aside-block",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "I know all the twitters.",
                },
                "ref": null,
                "rendered": "I know all the twitters.",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " - Ken Elliott",
                },
                "ref": null,
                "rendered": " - Ken Elliott",
                "type": "span",
              },
            ],
            "type": "div",
          },
          "type": "aside",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
